_core/admin/delpost.php

```diff
@@ -74,11 +74,8 @@ function targeted($resno, $pwd, $imgonly = 0, $automatic = 0, $children = 1, $di
             $auser   = $mysql->escape_string($_COOKIE['saguaro_auser']);
             $adfsize = ($row['fsize'] > 0) ? 1 : 0;
             $adname  = str_replace('</span> <span class="postertrip">!', '#', $row['name']);
-            if ($imgonly) {
-                $imgonly = 1;
-            } else {
-                $imgonly = 0;
-            }
+            $imgonly = ($imgonly =="on") ? 1 : 0;
```

Comment:
The hell is this?

Just pass it as a boolean to begin with. `on` is arbitrary enough as is.